src/sagas/featureToggle.js

```diff
@@ -102,86 +102,37 @@ export function* monitorFeatureFlags(currentRetry = 0) {
     },
   };
 
+  const unleashClient = new UnleashClient({
+    url: UNLEASH_URL,
+    clientKey: UNLEASH_CLIENT_KEY,
+    refreshInterval: -1,
+    disableRefresh: true, // Disable it, we will handle it ourselves
+    appName: `wallet-mobile-${Platform.OS}`,
+    context: options,
+  });
+
   try {
-    yield call(() => unleashClient.updateContext(options));
     yield put(setUnleashClient(unleashClient));
 
-    // Listeners should be set before unleashClient.start so we don't miss
-    // updates
-    yield fork(setupUnleashListeners, unleashClient);
-
-    // Start without awaiting it so we can listen for the
-    // READY event
-    unleashClient.start();
-
-    const { error, timeout } = yield race({
-      error: take(types.FEATURE_TOGGLE_ERROR),
-      success: take(types.FEATURE_TOGGLE_READY),
-      timeout: delay(CONNECT_TIMEOUT),
-    });
-
-    if (error || timeout) {
-      throw new Error('Error or timeout while connecting to unleash proxy.');
-    }
+    yield call(() => unleashClient.fetchToggles());
 
     // Fork the routine to download toggles.
     yield fork(fetchTogglesRoutine);
 
-    // At this point, unleashClient.start() already fetched the toggles
-    const featureToggles = mapFeatureToggles(unleashClient.toggles);
+    // At this point, unleashClient.fetchToggles() already fetched the toggles
+    // (this will throw if it hasn't)
+    const featureToggles = mapFeatureToggles(unleashClient.getToggles());
 
     yield put(setFeatureToggles(featureToggles));
     yield put(featureToggleInitialized());
   } catch (e) {
-    console.error('Error initializing unleash');
-    unleashClient.stop();
-
     yield put(setUnleashClient(null));
 
     // Wait 500ms before retrying
     yield delay(500);
 
     // Spawn so it's detached from the current thread
     yield spawn(handleInitFailed, currentRetry);
-  } finally {
-    if (yield cancelled()) {
-      yield call(() => unleashClient.stop());
-    }
-  }
-}
-
-export function* setupUnleashListeners(unleashClient) {
```

Comment:
How are you handling this now?

Reply:
In the `fetchToggleRoutine` saga, it checks if the toggles were updated and yields a `FEATURE_TOGGLE_UPDATE` action